### ISSUE TITLE
Create helpers for communicating loading states to a user

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#4390]
We offer various types of loading patterns (full page and skeleton) to allow designers and developers to create experiences that are incrementally loaded. While the meaning of these patterns may be obvious to sighted users, it is often not indicated to low/no-vision who may be using screen reader technology.

For example, if we need to load tiles as part of a dashboard we may use a collection of cards framed as a skeleton state while we wait for the data to load. Visually, it is clear that this part of the page is unavailable to a user. In this case, someone viewing a screen reader should also realize that this region of a page is unavailable and is currently loading.

When the data is loaded and the cards are drawn, the page becomes interactive. This status change should also be communicated to a user.

This issue is to determine the best pattern for indicating that regions of a page are being loaded, and communicate when they are loaded, to users who incorporate screen readers into their workflow.

**Comment by joshblack** [comment]
fyi @designertyler, might be related to navigation/loading patterns!

**Comment by joshblack** [comment]
One potential experience would be to use live regions to communicate when a region of a page is being loaded versus not:

[Screen Recording 2019-10-15 at 4.25.25 PM.zip](https://github.com/carbon-design-system/carbon/files/3744918/Screen.Recording.2019-10-15.at.4.25.25.PM.zip)

Code sample: https://codesandbox.io/s/inspiring-cerf-et098

---

Potential challenges:

- Would we only want to communicate the state of a region once? How could a user lookup the state of a specific region? (loading or available)
- Do we need to communicate the loaded state? What would this look like?
- Whenever a state changes for a region, should we announce all regions' states or just one that has changed?
